src/TuxArchitectStores.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+//architecture OBJECT: stores will register their outputs to this object so that store inputs can lookup those outputs later
+var architecture = {};
+
+//ArchitectChain CONSTRUCTOR FUNCTION: produces architectChain instances which allow stores to chain methods for defining their inputs and outputs
+//@param storeToArchitect: store OBJECT that inputs and outputs will be registered to
+var ArchitectChain = function (storeToArchitect) {
+  this.storeToArchitect = storeToArchitect;
+};
+
+//prototype methods
+//_getRegisterId FUNCTION: internal function,for getting the store __registerId__ from either the passed in store or the store mapped to the passed in input
+//@param inputOrStore STRING: input string to be checked against the architecture OBJECT for the corresponding store [ALTERNATE OBJECT: if a store is passed in than lookup __registerId__ on the passed in store]
+ArchitectChain.prototype._getRegisterId = function (inputOrStore) {
```

Comment:
Recommend grouping all private methods at the bottom of the file to provide separation.